lib/Listener/TaskProcessingListener.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Listener;
+
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Db\MessageMapper;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\TaskProcessing\Events\TaskSuccessfulEvent;
+use OCP\TaskProcessing\TaskTypes\TextToTextSummary;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @template-implements IEventListener<Event>
+ */
+class TaskProcessingListener implements IEventListener {
+
+	public function __construct(
+		private LoggerInterface $logger,
+		private MessageMapper $messageMapper,
+	) {
+	}
+
+	public function handle(Event $event): void {
+
+		if (!($event instanceof TaskSuccessfulEvent)) {
+			return;
+		}
+
+		$task = $event->getTask();
+
+		if ($task->getAppId() !== Application::APP_ID) {
+			return;
+		}
+
+		if ($task->getTaskTypeId() !== TextToTextSummary::ID) {
+			return;
+		}
+
+		if ($task->getCustomId() !== null) {
+			[$type, $id] = explode(':', $task->getCustomId());
+		} else {
+			$this->logger->info('Error handling task processing event custom id missing');
```

Comment:
For debuggability I suggest to log the value of the custom ID
```suggestion
			$this->logger->info('Error handling task processing event custom id missing', ['taskCustomId' => $task->getCustomid(), ]);
```

Reply:
Done